promql/engine.go

```diff
@@ -2028,6 +2073,25 @@ func (ev *evaluator) eval(ctx context.Context, expr parser.Expr) (parser.Value,
 				vec, err := ev.VectorscalarBinop(e.Op, v[1].(Vector), Scalar{V: v[0].(Vector)[0].F}, true, e.ReturnBool, enh, e.PositionRange())
 				return vec, handleVectorBinopError(err, e)
 			}, e.LHS, e.RHS)
+
+		case lt == parser.ValueTypeMatrix && rt == parser.ValueTypeScalar:
+			var allAnons annotations.Annotations
```

Comment:
Let's call it `allAnnos`, as `annos` is used as a variable name for annotations in other places in the engine. `Anons` (with one `n`) sounds like you're tracking anonymous things or something like that :)